xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAHD.cpp

```diff
@@ -814,3 +813,208 @@ bool CProcessorHD::IsBT2020Supported()
+
+    constexpr std::string_view rgb("RGB_");
+    if (DX::DXGIColorSpaceTypeToString(cs).compare(0, rgb.size(), rgb) == 0)
+      rgbColorSpaces.push_back(cs);
+
+    constexpr std::string_view ycbcr("YCBCR_");
+    if (DX::DXGIColorSpaceTypeToString(cs).compare(0, ycbcr.size(), ycbcr) == 0)
+      ycbcrColorSpaces.push_back(cs);
+  }
+
+  // Only probe the output formats of RGB/BGR type supported by the processor.
+  std::vector<DXGI_FORMAT> outputFormats;
+  for (const auto& format : GetProcessorOutputFormats())
+  {
+    std::string name = DX::DXGIFormatToString(format);
+    if (name.find('R') != std::string::npos && name.find('G') != std::string::npos &&
+        name.find('B') != std::string::npos)
+      outputFormats.push_back(format);
+  }
+
+  // Color spaces supported directly by the swap chain - as a set for easy lookup
+  std::vector<DXGI_COLOR_SPACE_TYPE> bbcs = DX::DeviceResources::Get()->SwapChainColorSpaces();
+  std::set<DXGI_COLOR_SPACE_TYPE> backbufferColorSpaces(bbcs.begin(), bbcs.end());
+
+  std::string conversions;
+
+  // The input format cannot be worked around and is fixed.
+  // Loop over the lists of:
+  // - input color spaces
+  // - output formats
+  // - output color spaces
+  for (const DXGI_COLOR_SPACE_TYPE& inputCS : ycbcrColorSpaces)
+  {
+    for (const DXGI_FORMAT& outputFormat : outputFormats)
+    {
+      for (const DXGI_COLOR_SPACE_TYPE& outputCS : rgbColorSpaces)
+      {
+        if (SUCCEEDED(m_pEnumerator1->CheckVideoProcessorFormatConversion(
+                inputFormat, inputCS, outputFormat, outputCS, &supported)) &&
+            supported == TRUE)
+        {
+          conversions.append("\n");
+          conversions.append(StringUtils::Format(
+              "{} {} / {}{} {:<{}} to {} {:<{}} / {}{} {:<{}}", "*",
+              DX::DXGIFormatToString(inputFormat),
+              (inputCS == heuristicsCS.inputColorSpace) ? "*" : " ",
+              (inputCS == inputNativeCS) ? "N" : " ", DX::DXGIColorSpaceTypeToString(inputCS), 32,
+              (outputFormat == heuristicsOutputFormat) ? "*" : " ",
+              DX::DXGIFormatToString(outputFormat), 26,
+              (outputCS == heuristicsCS.outputColorSpace) ? "*" : " ",
+              (backbufferColorSpaces.find(outputCS) != backbufferColorSpaces.end()) ? "bb" : "  ",
+              DX::DXGIColorSpaceTypeToString(outputCS), 32));
+        }
+      }
+    }
+  }
+
+  CLog::LogF(LOGDEBUG,
+             "supported conversions from format {}\n(*: values picked by "
+             "heuristics, N native input color space, bb supported as swap chain backbuffer){}",
+             DX::DXGIFormatToString(inputFormat), conversions);
+}
+
+std::vector<DXGI_FORMAT> CProcessorHD::GetProcessorOutputFormats() const
+{
+  std::vector<DXGI_FORMAT> result;
+
+  UINT uiFlags;
+  for (int fmt = DXGI_FORMAT_UNKNOWN; fmt <= DXGI_FORMAT_V408; fmt++)
+  {
+    if (S_OK == m_pEnumerator->CheckVideoProcessorFormat((DXGI_FORMAT)fmt, &uiFlags) &&
```

Comment:
Avoid C style cast, maybe create aux variable as is used later in `push_back()`

`DXGI_FORMAT format = static_cast<DXGI_FORMAT>(fmt);`